src/oemof/network/energy_system.py

```diff
@@ -168,17 +168,21 @@ def groups(self):
             (
                 g(n, gs)
                 for g in self._groupings
-                for n in self.nodes[self._first_ungrouped_node_index_ :]
+                for n in list(self.nodes)[self._first_ungrouped_node_index_ :]
             ),
             maxlen=0,
         )
         self._first_ungrouped_node_index_ = len(self.nodes)
         return self._groups
 
     @property
-    def nodes(self):
+    def node(self):
         return self._nodes
 
+    @property
+    def nodes(self):
+        return self._nodes.values()
+
     @nodes.setter
     def nodes(self, value):
```

Comment:
This should be adapted as well, as otherwise `_nodes` would get overwritten when using setter

Reply:
At least it is not consistent with getter

Reply:
Do we need the setter, anyway?

Reply:
That's a good question...

Reply:
My guess is that it is never used.